merlion/models/forecast/deep_models/deep_forecast_base.py

```diff
@@ -0,0 +1,340 @@
+#
+# Copyright (c) 2022 salesforce.com, inc.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+#
+"""
+    Base class for Deep Learning Forecasting Models
+"""
+import pdb
+import copy
+import time
+
+import logging
+import numpy as np
+import pandas as pd
+from scipy.stats import norm
+
+from typing import List, Optional, Tuple, Union
+from abc import abstractmethod
+
+try:
+    import torch
+    import torch.nn as nn
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[deep-learning]` or "
+        "`pip install `salesforce-merlion[all]`"
+    )
+    raise ImportError(str(e) + ". " + err)
+
+
+from merlion.models.deep_base import DeepConfig, DeepModelBase
+from merlion.models.forecast.base import ForecasterBase, ForecasterConfig
+from merlion.models.utils.rolling_window_dataset import RollingWindowDataset
+from merlion.utils.timefeatures import time_features
+
+from merlion.transform.base import TransformBase, Identity
+from merlion.transform.factory import TransformFactory
+from merlion.utils.misc import initializer
+from merlion.utils.time_series import to_pd_datetime, to_timestamp, TimeSeries, AggregationPolicy, MissingValuePolicy
+
+logger = logging.getLogger(__name__)
+
+
+class DeepForecasterConfig(DeepConfig, ForecasterConfig):
+    def __init__(
+        self,
+        n_past: int,
+        max_forecast_steps: int = None,
+        lr: float = 1e-3,
+        batch_size: int = 32,
+        num_epochs: int = 1,
+        optim_name: str = "adam",
+        criterion: str = "mse",
+        target_seq_index: int = None,
+        invert_transform=None,
+        start_token_len: int = 0,
+        ts_encoding: bool = True,
+        ts_freq: str = "h",
```

Comment:
There is no need to repeat arguments from the superclass's initializer, unless you want to change the default value. They will all be captured in `**kwargs`. I would also recommend moving `ts_encoding` / `ts_freq` to `DeepConfig`, since those arguments can be relevant for all deep models.